File: plugin/src/org/jetbrains/yesod/julius/JuliusFile.java
package org.jetbrains.yesod.julius;

/**
 * @author Leyla H
 */

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.lang.Language;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import org.jetbrains.annotations.NotNull;

public class JuliusFile extends PsiFileBase{
    public JuliusFile(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, JuliusLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public FileType getFileType() {
        return JuliusFileType.INSTANCE;
    }
}

File: plugin/src/org/jetbrains/yesod/julius/JuliusFile.kt
package org.jetbrains.yesod.julius

/**
 * @author Leyla H
 */

import com.intellij.extapi.psi.PsiFileBase
import com.intellij.lang.Language
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider

class JuliusFile(viewProvider: FileViewProvider) : PsiFileBase(viewProvider, JuliusLanguage.INSTANCE) {

    override fun getFileType(): FileType {
        return JuliusFileType.INSTANCE
    }
}
